### PR TITLE
Add WaveformLoader

### PR DESCRIPTION
**Pull request by rossant**
`WaveformLoader` is a class that extracts waveforms from traces.

### CODE REVIEW

**Comment by rossant** [comment]
Ready for review/merge @nippoo

File: phy/waveform/loader.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# Waveform loader from traces
+#------------------------------------------------------------------------------
+
+def _before_after(n_samples):
+    """Get the number of samples before and after."""
+    if not isinstance(n_samples, (tuple, list)):
+        before = n_samples // 2
+        after = n_samples - before
+    else:
+        before, after = n_samples
```
Comment (nippoo) [inline phy/waveform/loader.py]: Check to see whether ```len(n_samples) == 2``` ?

File: phy/waveform/loader.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    assert after >= 0
+    assert before + after == n_samples
+    return before, after
+
+
+def _slice(index, n_samples, margin=None):
+    """Return a waveform slice."""
+    if margin is None:
+        margin = (0, 0)
+    before, after = n_samples
```
Comment (nippoo) [inline phy/waveform/loader.py]: and check before here, I guess, too. For margin and n_samples?

File: phy/waveform/loader.py (hunk 1 above)
Comment (rossant) [inline phy/waveform/loader.py]: done

File: phy/waveform/loader.py (hunk 2 above)
Comment (rossant) [inline phy/waveform/loader.py]: done

**Comment by rossant** [comment]
Good for merge @nippoo ?

**Pull request merged**